### PR TITLE
fix: prevent modules from producing same id hash

### PR DESCRIPTION
**Pull request by digitalsadhu**
## JIRA Issue
[COREWEB-196](https://jira.finn.no/browse/COREWEB-196)

## Status
**READY**

## Description
There was an obscure bug where the source code of 2 feed items was
totally identical which, when hashed, produced the same ids. Since we
can't have 2 different modules with the same id hash, we needed to add
another parameter to the hashing. I've simply taken the previous id
value (that generated by browserify) which is the path to the file
and prepended it to content to be hashed.

See source content for object.entries and object.values:
https://github.com/es-shims/Object.entries/blob/master/index.js
https://github.com/es-shims/Object.values/blob/master/index.js

## Todos
- [x] Tests

## Deploy Notes
* [ ] merge/publish
* [ ] update `@asset-pipe/client` with the new version
* [ ] update `@finn-no/podlet` with new version of the client
* [ ] update all podlets with new version of `@finn-no/podlet`
* [ ] deploy everything

## Related PRs
* None

### CODE REVIEW

File: lib/browserify-plugins/id-hasher/id-hash-transform.js [hunk 1]
```diff
@@ -7,7 +7,10 @@ module.exports = function idHashTransform(idToHashMap) {
     return new Transform({
         objectMode: true,
         transform(obj, enc, next) {
-            idToHashMap[obj.id] = common.hasher(obj.source);
+            // include id and source in hash to avoid situation where
+            // multiple modules have the same source code but different
+            // depedendencies somewhere down the line.
+            idToHashMap[obj.id] = common.hasher(obj.id + obj.source);
```
Comment (SimenB) [inline lib/browserify-plugins/id-hasher/id-hash-transform.js]: `path.relatrive(__dirname, obj.source.path)` or something like that?

Comment (digitalsadhu) [reply lib/browserify-plugins/id-hasher/id-hash-transform.js]: yea, relative to process.cwd()? The scenario where determinism breaks down is where you have multiple asset servers on different machines trying to produce the same bundle. process.cwd would always contain the machine unique part that we want to trim out right? (Having trouble reasoning it out in my head)

File: lib/browserify-plugins/id-hasher/id-hash-transform.js [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
 module.exports = function idHashTransform(idToHashMap) {
     return new Transform({
         objectMode: true,
         transform(obj, enc, next) {
-            idToHashMap[obj.id] = common.hasher(obj.source);
+            // include id and source in hash to avoid situation where
+            // multiple modules have the same source code but different
+            // depedendencies somewhere down the line.
+            idToHashMap[obj.id] = common.hasher(
+                relative(process.cwd(), obj.id) + obj.source
```
Comment (SimenB) [inline lib/browserify-plugins/id-hasher/id-hash-transform.js]: `obj.id` is the path of the module?

Comment (digitalsadhu) [reply lib/browserify-plugins/id-hasher/id-hash-transform.js]: yeah, full path on the machine that built it

**Pull request merged**